inject-java/src/main/java/io/micronaut/annotation/processing/BeanDefinitionInjectProcessor.java

```diff
@@ -2245,15 +2245,14 @@ private ExecutableElementParamInfo populateParameterData(@Nullable String declar
                 switch (kind) {
                     case ARRAY:
                         ArrayType arrayType = (ArrayType) typeMirror;
-                        TypeMirror componentType = arrayType.getComponentType();
                         Object resolvedType = modelUtils.resolveTypeReference(arrayType);
                         params.addParameter(
                                 argName,
                                 resolvedType,
                                 genericUtils.resolveTypeReference(arrayType, boundTypes)
                         );
-                        params.addGenericTypes(argName, Collections.singletonMap("E", modelUtils.resolveTypeReference(componentType)));
-
+                        //TypeMirror componentType = arrayType.getComponentType();
+                        //params.addGenericTypes(argName, Collections.singletonMap("E", modelUtils.resolveTypeReference(componentType)));
```

Comment:
@graemerocher I couldn't understand why arrays were given a static generic E. It didn't make sense to me so I commented it out.